src/PIL/ImageShow.py

```diff
@@ -234,150 +235,80 @@ def show_file(self, path=None, **options):
                 path = options.pop("file")
             else:
                 raise TypeError("Missing required argument: 'path'")
-        subprocess.Popen(["xdg-open", path])
-        self._remove_path_after_delay(path)
+
+        path = quote(path)
+        command = self.get_command(path, **options)
+
+        kwargs = {
+            "stdout": subprocess.PIPE,
+        }
+
+        th = threading.Thread(
+            target=subprocess.run, args=(command.split(),), kwargs=kwargs, name=path
```

Comment:
We've recently had to address the possibility of [the file path having a space in it](https://github.com/python-pillow/Pillow/pull/3450#issuecomment-1009608418). This would `split()` incorrectly in such a situation, yes?

Reply:
1) I do not think that there is a a split problem, because the file(path) is quoted, and the path always be represented as a single entity.
2) I wouldn't be concerned about users having a temporary path "foo bar baz", I think that they will be able to sort this out.

Reply:
The solution was to run the viewer command as a shell command. User input has been sanitized in **get_command_ex** stage.  As a result (a) path problem solved; (b) possible security problem solved;
```python
        kwargs = {"shell": True, "stdout": subprocess.PIPE, "stderr": subprocess.PIPE}

        th = threading.Thread(
            target=subprocess.run, args=(command,), kwargs=kwargs, name=path
        )
```